Sources/GRPCCore/Streaming/Internal/BroadcastAsyncSequence.swift

```diff
@@ -682,16 +686,18 @@ struct _BroadcastSequenceStateMachine<Element: Sendable>: Sendable {
         _ continuation: ConsumerContinuation,
         forSubscription id: _BroadcastSequenceStateMachine<Element>.Subscriptions.ID
       ) -> OnSetContinuation {
-        let didSet = self.subscriptions.setContinuation(continuation, forSubscriber: id)
-        return didSet ? .none : .resume(continuation, .failure(CancellationError()))
+        if self.subscriptions.setContinuation(continuation, forSubscriber: id) {
+          return .none
+        } else {
+          return .resume(continuation, .failure(CancellationError()))
+        }
       }
 
       @inlinable
       mutating func cancel(
         _ id: _BroadcastSequenceStateMachine<Element>.Subscriptions.ID
       ) -> OnCancelSubscription {
-        let (removed, continuation) = self.subscriptions.removeSubscriber(withID: id)
-        assert(removed)
```

Comment:
Are we removing this because it would sometimes fail or just because it seemed unnecessary to assert it?